File: javascript/easy/word-count.js
export const countWords = (string) => {
  let object = {};
  string.match(/(?!')[a-z'0-9]{1,}(?<!')/gi).forEach((word) => {
    let lowercase = word.toLowerCase();
    if (object[lowercase]) {
      object[lowercase] += 1;
    } else {
      object = { ...object, [lowercase]: 1 };
    }
  });
  return object;
};

File: python/easy/word_count.py
import re

def count_words(sentence):
    split_sentence = re.sub("([^\w']|_)+", " ", sentence.strip("'")).lower().strip().split(" ")
    count = {}
    for word in split_sentence:
        clean_word = word.strip("'")
        if clean_word not in count:
            count[clean_word] = 1
        else:
            count[clean_word] += 1
    return count
